File: com.bosch.fsp.master_of_plow.gen/src/com/bosch/nevonex/fcal/impl/ACEINNA_ANGLERATE_T_R.java
/**
Copyright (c) Robert Bosch GmbH. All rights reserved.
*/
package com.bosch.nevonex.fcal.impl;

import com.bosch.nevonex.fcal.IACEINNA_ANGLERATE_T_R;

import org.eclipse.emf.ecore.EClass;
import org.eclipse.emf.ecore.EObject;
import org.eclipse.emf.ecore.InternalEObject;

import org.eclipse.emf.ecore.impl.EObjectImpl;

/**
 * @generated
 */
public class ACEINNA_ANGLERATE_T_R extends EObjectImpl implements IACEINNA_ANGLERATE_T_R {
	protected float gyroY = 0.0f;
	protected float gyroX = 0.0f;
	protected float gyroZ = 0.0f;
	protected int gyroY_FigureOfMerit = 0;
	protected int gyroX_FigureOfMerit = 0;
	protected int gyroZ_FigureOfMerit = 0;
	protected float angleRate_Latency = 0.0f;

	protected ACEINNA_ANGLERATE_T_R() {
		super();
	}

	@Override
	protected EClass eStaticClass() {
		return FcalPackage.Literals.ACEINNA_ANGLERATE_T_R;
	}

	public float getGyroY() {
		return gyroY;
	}

	public void setGyroY(float newGyroY) {
		gyroY = newGyroY;
	}

	public float getGyroX() {
		return gyroX;
	}

	public void setGyroX(float newGyroX) {
		gyroX = newGyroX;
	}

	public float getGyroZ() {
		return gyroZ;
	}

	public void setGyroZ(float newGyroZ) {
		gyroZ = newGyroZ;
	}

	public int getGyroY_FigureOfMerit() {
		return gyroY_FigureOfMerit;
	}

	public void setGyroY_FigureOfMerit(int newGyroY_FigureOfMerit) {
		gyroY_FigureOfMerit = newGyroY_FigureOfMerit;
	}

	public int getGyroX_FigureOfMerit() {
		return gyroX_FigureOfMerit;
	}

	public void setGyroX_FigureOfMerit(int newGyroX_FigureOfMerit) {
		gyroX_FigureOfMerit = newGyroX_FigureOfMerit;
	}

	public int getGyroZ_FigureOfMerit() {
		return gyroZ_FigureOfMerit;
	}

	public void setGyroZ_FigureOfMerit(int newGyroZ_FigureOfMerit) {
		gyroZ_FigureOfMerit = newGyroZ_FigureOfMerit;
	}

	public float getAngleRate_Latency() {
		return angleRate_Latency;
	}

	public void setAngleRate_Latency(float newAngleRate_Latency) {
		angleRate_Latency = newAngleRate_Latency;
	}

	@Override
	public Object eGet(int featureID, boolean resolve, boolean coreType) {
		switch (featureID) {
		case FcalPackage.ACEINNA_ANGLERATE_T_R__GYRO_Y:
			return getGyroY();
		case FcalPackage.ACEINNA_ANGLERATE_T_R__GYRO_X:
			return getGyroX();
		case FcalPackage.ACEINNA_ANGLERATE_T_R__GYRO_Z:
			return getGyroZ();
		case FcalPackage.ACEINNA_ANGLERATE_T_R__GYRO_Y_FIGURE_OF_MERIT:
			return getGyroY_FigureOfMerit();
		case FcalPackage.ACEINNA_ANGLERATE_T_R__GYRO_X_FIGURE_OF_MERIT:
			return getGyroX_FigureOfMerit();
		case FcalPackage.ACEINNA_ANGLERATE_T_R__GYRO_Z_FIGURE_OF_MERIT:
			return getGyroZ_FigureOfMerit();
		case FcalPackage.ACEINNA_ANGLERATE_T_R__ANGLE_RATE_LATENCY:
			return getAngleRate_Latency();
		}
		return super.eGet(featureID, resolve, coreType);
	}

	@Override
	public void eSet(int featureID, Object newValue) {
		switch (featureID) {
		case FcalPackage.ACEINNA_ANGLERATE_T_R__GYRO_Y:
			setGyroY((Float) newValue);
			return;
		case FcalPackage.ACEINNA_ANGLERATE_T_R__GYRO_X:
			setGyroX((Float) newValue);
			return;
		case FcalPackage.ACEINNA_ANGLERATE_T_R__GYRO_Z:
			setGyroZ((Float) newValue);
			return;
		case FcalPackage.ACEINNA_ANGLERATE_T_R__GYRO_Y_FIGURE_OF_MERIT:
			setGyroY_FigureOfMerit((Integer) newValue);
			return;
		case FcalPackage.ACEINNA_ANGLERATE_T_R__GYRO_X_FIGURE_OF_MERIT:
			setGyroX_FigureOfMerit((Integer) newValue);
			return;
		case FcalPackage.ACEINNA_ANGLERATE_T_R__GYRO_Z_FIGURE_OF_MERIT:
			setGyroZ_FigureOfMerit((Integer) newValue);
			return;
		case FcalPackage.ACEINNA_ANGLERATE_T_R__ANGLE_RATE_LATENCY:
			setAngleRate_Latency((Float) newValue);
			return;
		}
		super.eSet(featureID, newValue);
	}

	@Override
	public void eUnset(int featureID) {
		switch (featureID) {
		case FcalPackage.ACEINNA_ANGLERATE_T_R__GYRO_Y:
			setGyroY(0.0f);
			return;
		case FcalPackage.ACEINNA_ANGLERATE_T_R__GYRO_X:
			setGyroX(0.0f);
			return;
		case FcalPackage.ACEINNA_ANGLERATE_T_R__GYRO_Z:
			setGyroZ(0.0f);
			return;
		case FcalPackage.ACEINNA_ANGLERATE_T_R__GYRO_Y_FIGURE_OF_MERIT:
			setGyroY_FigureOfMerit(0);
			return;
		case FcalPackage.ACEINNA_ANGLERATE_T_R__GYRO_X_FIGURE_OF_MERIT:
			setGyroX_FigureOfMerit(0);
			return;
		case FcalPackage.ACEINNA_ANGLERATE_T_R__GYRO_Z_FIGURE_OF_MERIT:
			setGyroZ_FigureOfMerit(0);
			return;
		case FcalPackage.ACEINNA_ANGLERATE_T_R__ANGLE_RATE_LATENCY:
			setAngleRate_Latency(0.0f);
			return;
		}
		super.eUnset(featureID);
	}

	@Override
	public boolean eIsSet(int featureID) {
		switch (featureID) {
		case FcalPackage.ACEINNA_ANGLERATE_T_R__GYRO_Y:
			return gyroY != 0.0f;
		case FcalPackage.ACEINNA_ANGLERATE_T_R__GYRO_X:
			return gyroX != 0.0f;
		case FcalPackage.ACEINNA_ANGLERATE_T_R__GYRO_Z:
			return gyroZ != 0.0f;
		case FcalPackage.ACEINNA_ANGLERATE_T_R__GYRO_Y_FIGURE_OF_MERIT:
			return gyroY_FigureOfMerit != 0;
		case FcalPackage.ACEINNA_ANGLERATE_T_R__GYRO_X_FIGURE_OF_MERIT:
			return gyroX_FigureOfMerit != 0;
		case FcalPackage.ACEINNA_ANGLERATE_T_R__GYRO_Z_FIGURE_OF_MERIT:
			return gyroZ_FigureOfMerit != 0;
		case FcalPackage.ACEINNA_ANGLERATE_T_R__ANGLE_RATE_LATENCY:
			return angleRate_Latency != 0.0f;
		}
		return super.eIsSet(featureID);
	}
} //ACEINNA_ANGLERATE_T_R

File: master_of_plow_CPP_SDK/src-gen/nevonex/fcal/ACEINNA_ANGLERATE_T_R.cpp
/*
 * nevonex/fcal/ACEINNA_ANGLERATE_T_R.cpp
 * Copyright (c) Robert Bosch GmbH. All rights reserved.
 */

#include "ACEINNA_ANGLERATE_T_R.hpp"
#include <ecore/EObject.hpp>
#include <ecore/EClass.hpp>
#include "nevonex/fcal/FcalPackage.hpp"
#include <ecorecpp/mapping.hpp>
#include <algorithm>
#include <nevonex/fcb/SubscriberEnum.hpp>
#include <nevonex-fcal-platform/log/Logger.hpp>

#if BOOST_VERSION >= 106501
#define BOOST_STACKTRACE_HEADERS_FOUND
#include <boost/stacktrace.hpp>
#endif

/*PROTECTED REGION ID(ACEINNA_ANGLERATE_T_R.cpp) START*/
// Please, enable the protected region if you add manually written code.
// To do this, add the keyword ENABLED before START.
/*PROTECTED REGION END*/

using namespace ::nevonex::fcal;
using namespace ::nevonex::log;

// Default constructor
ACEINNA_ANGLERATE_T_R::ACEINNA_ANGLERATE_T_R() : m_gyroY(0.0f), m_gyroX(0.0f), m_gyroZ(0.0f), m_gyroY_FigureOfMerit(0), m_gyroX_FigureOfMerit(0), m_gyroZ_FigureOfMerit(0), m_angleRate_Latency(0.0f)
{

    /*PROTECTED REGION ID(ACEINNA_ANGLERATE_T_R__ACEINNA_ANGLERATE_T_R) START*/
    // Please, enable the protected region if you add manually written code.
    // To do this, add the keyword ENABLED before START.
    /*PROTECTED REGION END*/

}

ACEINNA_ANGLERATE_T_R::~ACEINNA_ANGLERATE_T_R()
{

    /*PROTECTED REGION ID(ACEINNA_ANGLERATE_T_R__Destructor) START*/
    // Please, enable the protected region if you add manually written code.
    // To do this, add the keyword ENABLED before START.
    /*PROTECTED REGION END*/

}

// Attributes

::ecore::EFloat ACEINNA_ANGLERATE_T_R::getGyroY() const
{
    return m_gyroY;
}

void ACEINNA_ANGLERATE_T_R::setGyroY(::ecore::EFloat _gyroY)
{

    ::ecore::EFloat _old_gyroY = m_gyroY;
    m_gyroY = _gyroY;

}

::ecore::EFloat ACEINNA_ANGLERATE_T_R::getGyroX() const
{
    return m_gyroX;
}

void ACEINNA_ANGLERATE_T_R::setGyroX(::ecore::EFloat _gyroX)
{

    ::ecore::EFloat _old_gyroX = m_gyroX;
    m_gyroX = _gyroX;

}

::ecore::EFloat ACEINNA_ANGLERATE_T_R::getGyroZ() const
{
    return m_gyroZ;
}

void ACEINNA_ANGLERATE_T_R::setGyroZ(::ecore::EFloat _gyroZ)
{

    ::ecore::EFloat _old_gyroZ = m_gyroZ;
    m_gyroZ = _gyroZ;

}

::ecore::EInt ACEINNA_ANGLERATE_T_R::getGyroY_FigureOfMerit() const
{
    return m_gyroY_FigureOfMerit;
}

void ACEINNA_ANGLERATE_T_R::setGyroY_FigureOfMerit(::ecore::EInt _gyroY_FigureOfMerit)
{

    ::ecore::EInt _old_gyroY_FigureOfMerit = m_gyroY_FigureOfMerit;
    m_gyroY_FigureOfMerit = _gyroY_FigureOfMerit;

}

::ecore::EInt ACEINNA_ANGLERATE_T_R::getGyroX_FigureOfMerit() const
{
    return m_gyroX_FigureOfMerit;
}

void ACEINNA_ANGLERATE_T_R::setGyroX_FigureOfMerit(::ecore::EInt _gyroX_FigureOfMerit)
{

    ::ecore::EInt _old_gyroX_FigureOfMerit = m_gyroX_FigureOfMerit;
    m_gyroX_FigureOfMerit = _gyroX_FigureOfMerit;

}

::ecore::EInt ACEINNA_ANGLERATE_T_R::getGyroZ_FigureOfMerit() const
{
    return m_gyroZ_FigureOfMerit;
}

void ACEINNA_ANGLERATE_T_R::setGyroZ_FigureOfMerit(::ecore::EInt _gyroZ_FigureOfMerit)
{

    ::ecore::EInt _old_gyroZ_FigureOfMerit = m_gyroZ_FigureOfMerit;
    m_gyroZ_FigureOfMerit = _gyroZ_FigureOfMerit;

}

::ecore::EFloat ACEINNA_ANGLERATE_T_R::getAngleRate_Latency() const
{
    return m_angleRate_Latency;
}

void ACEINNA_ANGLERATE_T_R::setAngleRate_Latency(::ecore::EFloat _angleRate_Latency)
{

    ::ecore::EFloat _old_angleRate_Latency = m_angleRate_Latency;
    m_angleRate_Latency = _angleRate_Latency;

}

// References

